src/components/marker-chart/Canvas.js

```diff
@@ -265,35 +273,37 @@ class MarkerChartCanvas extends React.PureComponent<Props, State> {
       if (rowIndex > 0 && name === markerTimingRows[rowIndex - 1].name) {
         continue;
       }
-      const displayedName =
-        name.length < MARKER_LABEL_MAX_LENGTH
-          ? name
-          : name.slice(0, MARKER_LABEL_MAX_LENGTH) + '…';
+      const fittedText = textMeasurement.getFittedText(name, MARGIN_LEFT);
       const y = rowIndex * rowHeight - viewportTop;
-      ctx.fillText(displayedName, 5, y + TEXT_OFFSET_TOP);
+      ctx.fillText(fittedText, 5, y + TEXT_OFFSET_TOP);
     }
   }
 
   hitTest = (x: CssPixels, y: CssPixels): IndexIntoMarkerTiming | null => {
+    if (x < MARGIN_LEFT) {
+      return null;
+    }
     const {
       rangeStart,
       rangeEnd,
       markerTimingRows,
       rowHeight,
       viewport: { viewportLeft, viewportRight, viewportTop, containerWidth },
     } = this.props;
+    const markerContainerWidth = containerWidth - MARGIN_LEFT - MARGIN_RIGHT;
 
     const rangeLength: Milliseconds = rangeEnd - rangeStart;
     const viewportLength: UnitIntervalOfProfileRange =
       viewportRight - viewportLeft;
     const unitIntervalTime: UnitIntervalOfProfileRange =
-      viewportLeft + viewportLength * (x / containerWidth);
+      viewportLeft +
+      viewportLength * ((x - MARGIN_LEFT) / markerContainerWidth);
     const time: Milliseconds = rangeStart + unitIntervalTime * rangeLength;
     const rowIndex = Math.floor((y + viewportTop) / rowHeight);
     const minDuration =
       rangeLength *
       viewportLength *
-      (rowHeight * 2 * MARKER_DOT_RADIUS / containerWidth);
+      (rowHeight * 2 * MARKER_DOT_RADIUS / markerContainerWidth);
```

Comment:
I think this `minDuration` isn't taken into account for markers that are "cut". Eg still the same one at the top left in https://deploy-preview-1286--perf-html.netlify.com/public/dfc1b84f9982b4d67e0c1079bad2c582cde1f32f/marker-chart/?globalTrackOrder=0-1-2-3&hiddenGlobalTracks=1-2-3&hiddenLocalTracksByPid=2399-0-1&localTrackOrderByPid=2399-0-1~2422-0~&thread=0&v=3&markerSearch=BHR

This marker is quite long, but because we see only the end of it, it's drawn as a dot. But the hit test algorithm doesn't find it until we reach the actual "end" of the "ghost marker" that would be here if we wouldn't cut it.

I think that the algorithm that uses `minDuration` below should check if `markerTiming.start[i]` is before/after the start of the chart, and use this to calculate the `end`.

This was here before your patch, but your patch makes it more visible: previously these markers were barely visible because they were below the nearly-opaque part of the label's gradient.

Reply:
I changed the rendering behavior to now use an `uncut` length to determine if something is a dot or not, so things longer bars don't turn into dots at the edges.